src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1170,6 +1178,26 @@ struct npc_silverpine_sylvanas_windrunner_high_command : public ScriptedAI
         if (Creature* forsakenWarhorse = me->FindNearestCreature(NPC_FORSAKEN_WARHORSE, 5.0f, true))
             forsakenWarhorse->CastSpell(forsakenWarhorse, me->HasAura(SPELL_APPLY_INVIS_ZONE_1) ? SPELL_APPLY_INVIS_ZONE_1 : SPELL_APPLY_INVIS_ZONE_4, true);
     }
+
+    void OnQuestAccept(Player* player, Quest const* quest) override
+    {
+        switch (quest->GetQuestId())
+        {
+            case QUEST_LORDAERON:
+                player->CastSpell(player, SPELL_SUMMON_SYLVANAS_AND_HORSE, true);
+                player->CastSpell(player, SPELL_SUMMON_FORSAKEN_WARHORSE, true);
+                player->CastSpell(player, SPELL_LORDAERON_AURA, true);
+                player->CastSpell(player, SPELL_SUMMON_LORDAERON_ACTORS, true);
```

Comment:
>```cpp
>    SPELL_LORDAERON_AURA                        = 84189, // Note: NYI, after researching thoroughly, nothing seems to point what this SPELL_AURA_DUMMY should do. Maybe it is what forces the player to cast the rest of the spells.
>```

I'm almost sure SourceSpellID (spell to cast on quest accept) is a thing on retail, despite it's not sent to client because not needed by client. That is easiest way to cast spells comparing to other possible ways. But what is done here I have no idea. Scripted auras for such stuff usually has small duration, this one infinite and if you preserved orded of casts, it does something else. All those spells also are casted by player, so maybe there's serverside master spell

Btw what happens if quest is abandoned and taken again? Player casts again all those spells? Using SourceSpellID raises question how to add condition for [this](https://tcubuntu.northeurope.cloudapp.azure.com/aowow/?npc=20130), which is much more easier case
When this aura gets removed btw?

Reply:
I checked the serverside spell list that we have access to and there's no master spell related to any of these at all. I checked around 250 ids before and after these, and there's nothing. I need to review the packets being sent because right now I'm not sure if I really preserved the order for this exact spell.

If the quest is retaken, all of the spells are recast again. The aura gets removed either when the player dismounts from the horse, which causes a skip to happen, and then the given complete quest spell is cast, or after that complete quest spell is cast.

Reply:
I just rechecked the packets and they follow this order in the same world tick:

SMSG_SPELL_START - 84128 (Summon Sylvanas + Horse)
SMSG_SPELL_START - 84164 (Summon Forsaken Warhorse)
SMSG_SPELL_START - 85406 (Detect: Quest Invis Zone 20 (Ambermill Shield)) [This one is handled by area_spell]
SMSG_SPELL_START - 84127 (Summon Lordaeron Actors)
SMSG_SPELL_START - 84189 (Lordaeron Area Aura)
SMSG_SPELL_START - 46598 (Ride Vehicle Hardcoded)
SMSG_SPELL_GO - 84189 (Lordaeron Area Aura)
SMSG_SPELL_GO - 84128 (Summon Sylvanas + Horse)
SMSG_SPELL_GO - 85406 (Detect: Quest Invis Zone 20 (Ambermill Shield))
SMSG_SPELL_GO - 84164 (Summon Forsaken Warhorse)
SMSG_SPELL_GO - 84127 (Summon Lordaeron Actors)
SMSG_SPELL_GO - 46598 (Ride Vehicle Hardcoded)

Reply:
It's exactly what I usually see, master spell is sent as last

Reply:
So are we to assume that aura spell dummy is the masterspell for everything else?

Reply:
>The aura gets removed either when the player dismounts from the horse, which causes a skip to happen, and then the given complete quest spell is cast, or after that complete quest spell is cast.

So it serves some kind of purpose, because of that it should be an aura with infinite duration

Reply:
Feedback applied!

Reply:
I've done more research on the packets, apparently, the aura is first applied and cast when No Escape quest is rewarded. It is on until Lordaeron quest is rewarded, so it cannot be the master spell for all of the rest.

Reply:
Any way, I have no idea how to use SourceSpellID for player cast, didn't encountered something like that in 335 yet

Reply:
SourceSpellId in master is contained in quest_template_addon, but if I remember correctly that value is taken directly from the sniff if there's any difference between the sniff and the DB and I had no update row.